Packet++/src/SomeIpSdLayer.cpp

```diff
@@ -636,28 +636,35 @@ bool SomeIpSdLayer::isDataValid(const uint8_t* data, size_t dataLen)
 	if (!data ||
 		dataLen < sizeof(someipsdhdr) + sizeof(uint32_t) ||
 		dataLen < sizeof(someipsdhdr) + sizeof(uint32_t) + getLenEntries(data) + sizeof(uint32_t) ||
-		dataLen < be32toh(*((uint32_t *)(data + sizeof(someipsdhdr) + sizeof(uint32_t) + getLenEntries(data)))))
+		dataLen < sizeof(someipsdhdr) + sizeof(uint32_t) + getLenEntries(data) + sizeof(uint32_t) + getLenOptions(data) ||
+		!countOptions(nullptr, data))
 	{
 		return false;
 	}
 
 	return true;
 }
 
-uint32_t SomeIpSdLayer::countOptions()
+bool SomeIpSdLayer::countOptions(uint32_t* count, const uint8_t* data)
```

Comment:
Instead of `uint32_t*` I'd put a reference `uint32_t&`, that's a more standard way in c++ to return a value.

Then in `isDataValid()` we can create a dummy variable to pass to this method